frontend/packages/ceph-storage-plugin/src/components/ocs-install/ipi-deployment-modal.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+
+import {
+  createModalLauncher,
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+} from '@console/internal/components/factory/modal';
+import {
+  PromiseComponent,
+  PromiseComponentState,
+} from '@console/internal/components/utils/promise-component';
+
+class IPIDeployementModal extends PromiseComponent<ipiModalProps, PromiseComponentState> {
```

Comment:
The `PromiseComponentState` type is currently used only in `promise-component.spec.tsx`.

I think that each modal component should explicitly declare its own state type, i.e. please add this after the props type:

```
type IPIDeployementModalState = {
  inProgress: boolean;
  errorMessage: string;
};
```